google/cloud/aiplatform/training_jobs.py

```diff
@@ -743,12 +778,27 @@ def _run_job(
             bigquery_destination=bigquery_destination,
         )
 
+        parent_model = models.ModelRegistry._get_true_version_parent(
+            parent_model=parent_model,
+            project=self.project,
+            location=self.location,
+        )
+
+        if model:
```

Comment:
It seems like this should go into `_prepare_and_validate_run` because it's coupled with the creation of the model proto.

Reply:
Preference is to leave it as is because _prepare_and_validate_run doesn't run for the AutoML training jobs and this keeps the versioning logic in one spot.

Open to pushback, however.

Reply:
SGTM